src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -183,12 +183,29 @@ public String prefix() throws IOException {
         }
     }
 
-    // @todo #15:30min Implement bnf() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public long bnf() {
-        throw new UnsupportedOperationException("bnf() not yet implemented");
+    public long bnf() throws IOException {
```

Comment:
@Vatavuk Let's change `bnf()` return type to `String`. As per whitepaper spec, the ID can be any arbitrary hex number, so it's not really numeric except for the format.

Note that in the Ruby, the String representation of `bnf` does not parse it as a number:
https://github.com/zold-io/zold/blob/e882a747c10e9fae83f4c4968c4882ad7590ba1c/lib/zold/txn.rb#L83